redaxo/src/addons/media_manager/lib/effects/effect_img2img.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/**
+ * @package redaxo\media-manager
+ */
+
+class rex_effect_img2img extends rex_effect_abstract
+{
+    private static $convert_types = [
+        'jpg',
+        'png',
+        'gif',
+        'webp',
+    ];
+
+    private static $convert_to = [
+        'jpg' => [
+            'ext' => 'jpg',
+            'content-type' => 'image/jpeg',
+        ],
+        'png' => [
+            'ext' => 'png',
+            'content-type' => 'image/png',
+        ],
+        'gif' => [
+            'ext' => 'gif',
+            'content-type' => 'image/gif',
+        ],
+        'webp' => [
+            'ext' => 'webp',
+            'content-type' => 'image/webp',
+        ],
+    ];
+
+    private static $convert_tos = ['jpg', 'png', 'gif', 'webp'];
+    private static $convert_to_default = 'webp';
+
+    public function execute()
+    {
+        $media = $this->media;
+
+        $ext = $media->getFormat();
+        $ext = 'jpeg' === $ext ? 'jpg' : $ext;
+        // skip if extension is not in list
+        if (!in_array(strtolower($ext), self::$convert_types)) {
+            return;
+        }
+
+        $media->asImage();
+        $imageObject = $media->getImage();
+        if (null === $imageObject) {
+            return;
+        }
+
+        if (!isset(self::$convert_to[$this->params['convert_to']])) {
+            $convert_to = self::$convert_to[self::$convert_to_default];
+        } else {
+            $convert_to = self::$convert_to[$this->params['convert_to']];
+        }
+
+        switch ($convert_to['ext']) {
+            case 'webp':
+                imagepalettetotruecolor($imageObject); // Prevent error 'Paletter image not supported by webp' (PNG mit indizierten Farben)
+                break;
+
+             case 'gif':
+                $w = $media->getWidth();
+                $h = $media->getHeight();
+
+                $transparencyColor = ['red' => 1, 'green' => 2, 'blue' => 3];
```

Comment:
Ich habe jetzt mal nahezu schwarz für die gif-Transparenzeingestellt. Bei hellem Hintergrund gibt es an den Rändern dann dunkle Pixel.

Reply:
Also wenn gif hier so problematisch ist, bin ich ja fast dazu geneigt, gif als zielformat doch wegzulassen, bis jemand ganz konkret es benötigt.

Reply:
Also ich würde es jetzt mal so drin lassen da GIF wahrscheinlich eher selten benutzt wird.

Würde eher sagen wenn mehrere sich noch zusätzlich die Verbesserung wünschen dann evtl. noch hinzufügen.

Wollte gar nicht so viel Zeit da reinstecken da ich nur webp-Format brauchte und @bitshiftersgmbh da schon was fertig hatte :)) 
... ok ist jetzt ganz anders geworden aber es funzt.

Und ich habe reichlich über den MM gelernt 👍

Reply:
Das betrifft ja nur gif-Format wenn von z.B. transparentem png transformiert wird ...

Reply:
Ja, ist für mich auch in Ordnung.